sphinx/domains/python.py

```diff
@@ -285,12 +290,54 @@ def add_target_and_index(self, name_cls, sig, signode):
                                               fullname, '', None))
 
     def before_content(self):
-        # needed for automatic qualification of members (reset in subclasses)
-        self.clsname_set = False
+        # type: () -> None
+        """Handle object nesting before content
+
+        If this class is a nestable object, such as a class object, build up a
```

Comment:
I feel a bit strange for "this class". How about "this instance" or "the instance"?